stock_intercompany/models/stock_picking.py

```diff
@@ -0,0 +1,33 @@
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def create_counterpart_picking(self):
+        companies = self.env["res.company"].sudo().search([])
+        partners = {cp.partner_id: cp for cp in companies}
+        if self.partner_id in partners.keys():
+            company = partners[self.partner_id]
+            warehouse = (
+                self.env["stock.warehouse"]
+                .sudo()
+                .search([("company_id", "=", company.id)], limit=1)
+            )
```

Comment:
I think it is better to use the warehouse of the picking type we configured in the res.config.settings:
```suggestion
            if `company.intercompany_in_type_id.warehouse_id:`
                warehouse = company.intercompany_in_type_id.warehouse_id
            else:
                warehouse = (
                    self.env["stock.warehouse"]
                    .sudo()
                    .search([("company_id", "=", company.id)], limit=1)
                )
```